src/Classes/MyRadio/CoreUtils.php

```diff
@@ -233,6 +233,42 @@ public static function makeInterval($start, $end)
     }
 
     /**
+     * Runs the relevant encode commands on an uploaded music file.
+     *
+     * @param string $tmpfile The original unencoded filepath
+     * @param string $dbfile  The destination filepath, sans extension
+     * @throws MyRadioException Thrown if encode or move commands appear to fail.
+     * @note Similar command is run for podcast uploads, which are done slightly differently
+     */
+    public static function encodeTrack($tmpfile, $dbfile)
+    {
+        $commands = [
+            'mp3' => "nice -n 15 ffmpeg -i '{$tmpfile}' -ab 192k -f mp3 -map 0:a '{$dbfile}.mp3'",
+            'ogg' => "nice -n 15 ffmpeg -i '{$tmpfile}' -acodec libvorbis -ab 192k -map 0:a '{$dbfile}.ogg'"
+        ];
+        $escaped_commands = array_map('escapeshellcmd', $commands);
+        $failed_formats = [];
+
+        foreach ($escaped_commands as $format => $command) {
+            exec($command, $command_stdout, $command_exit_code);
```

Comment:
http://php.net/manual/en/function.exec.php - these are passed by reference and therefore defined by exec.